frontend/taipy/src/ScenarioDag.tsx

```diff
@@ -133,14 +142,23 @@ const ScenarioDag = (props: ScenarioDagProps) => {
         if (displayModel) {
             setScenarioId(displayModel[0]);
             // populate model
-            doLayout = populateModel(displayModel, model);
+            doLayout = populateModel(addStatusToDisplayModel(displayModel, taskStatuses), model);
+        }
+        const rects = engine.getModel() && engine
+            .getModel()
+            .getNodes()
+            .reduce((pv, nm) => {
+                pv[nm.getID()] = nm.getPosition();
+                return pv;
+            }, {} as Record<string, Point>);
+        const hasPos = rects && Object.keys(rects).length;
+        if (hasPos) {
+            model.getNodes().forEach(nm => rects[nm.getID()] && nm.setPosition(rects[nm.getID()]));
         }
         engine.setModel(model);
-        // Block deletion
-        //engine.getActionEventBus().registerAction(new DeleteItemsAction({ keyCodes: [1] }));
         model.setLocked(true);
-        doLayout && setTimeout(relayout, 500);
-    }, [displayModel, engine, relayout, onClick]);
+        doLayout && !hasPos && setTimeout(relayout, 500);
```

Comment:
500ms is usually pretty slow for a user interaction.
Can this be smaller?

Reply:
Not sure
I remember trying ...